backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/applicationGraphQLParams.kt

```diff
@@ -0,0 +1,32 @@
+package app.ehrenamtskarte.backend.application.webservice
+
+import app.ehrenamtskarte.backend.application.webservice.schema.create.UploadKey
+import app.ehrenamtskarte.backend.common.webservice.GraphQLParams
+import com.expediagroup.graphql.SchemaGeneratorConfig
+import com.expediagroup.graphql.TopLevelObject
+import com.expediagroup.graphql.hooks.SchemaGeneratorHooks
+import graphql.kickstart.servlet.apollo.ApolloScalars
+import org.dataloader.DataLoaderRegistry
+import kotlin.reflect.KType
+import graphql.schema.*
+
+
+private fun createDataLoaderRegistry(): DataLoaderRegistry {
+    val dataLoaderRegistry = DataLoaderRegistry()
+    return dataLoaderRegistry
+}
+
+val applicationGraphQlParams = GraphQLParams(
+    config = SchemaGeneratorConfig(
+        supportedPackages = listOf("app.ehrenamtskarte.backend.application.webservice.schema"),
+        hooks = object : SchemaGeneratorHooks {
+            override fun willGenerateGraphQLType(type: KType): GraphQLType? =
+                if (type.classifier == UploadKey::class) ApolloScalars.Upload else null
+        }
+    ),
```

Comment:
Fancy. What does it do? 😄

Reply:
In general the GraphQL types are derived here from Kotlin data classes. According to the docs willGenerateGraphQLType intercepts the generation process if one would like to handle some types separately: 

_willGenerateGraphQLType_

_graphql.schema.GraphQLType willGenerateGraphQLType(kotlin.reflect.KType type)_

_Called before using reflection to generate the graphql object type for the given KType. This allows supporting objects that the caller does not want to use reflection on for special handling_

Reply:
Thanks @aiquita. Although I implemented, I did not really know, I just looked at some examples :D
Generally, we tell graphql to use the "standard" apollo signature (Upload) for adding files to a graphql request (as in https://www.apollographql.com/docs/apollo-server/data/file-uploads/)